src/libs/Violations/ViolationsUtils.ts

```diff
@@ -49,31 +49,55 @@ function getTagViolationsForSingleLevelTags(
 }
 
 /**
- * Calculates some tag levels required and missing tag violations for the given transaction
+ * Calculates missing tag violations for policies with dependent tags,
+ * by returning one per tag with its corresponding tagName in the data
  */
-function getTagViolationsForMultiLevelTags(
-    updatedTransaction: Transaction,
-    transactionViolations: TransactionViolation[],
-    policyRequiresTags: boolean,
-    policyTagList: PolicyTagList,
-): TransactionViolation[] {
+function getTagViolationsForDependentTags(policyTagList: PolicyTagList, transactionViolations: TransactionViolation[], tagName: string | undefined) {
```

Comment:
```suggestion
function getTagViolationsForDependentTags(policyTagList: PolicyTagList, transactionViolations: TransactionViolation[], tagName: string) {
```

Can we make `tagName` a string and fall back to empty string in the place where we call this?